### PR TITLE
Extend Badge component

### LINKED ISSUE #329: Improve Badge component to support other positions (i.e. centered ones)
# Description

> We stuck with a case when `Badge` is centered on top. So I think Badge component should be improved to support all possible positions. I think it should support centered case on every side and corners as well for each side. Feel free to put your ideas if have any...

### PR DESCRIPTION
**Pull request by yurist38**
# What does this PR do:
* Closes #329 
* Introduced new prop `align` into `<Badge />` component. It accepts one of these 3 values: [`start`, `center`, `end`]. The default behavior is not changed and this new prop is optional.
* Markdown of the `<Badge />` component is refactored to be more user-friendly and nice. 😄 
* Made badge is more responsive: now left and right badges have limitation by width and will not exceed 100% of the container width. Also, arrows were improved and became really responsive (this approach already used in `Hotels` widget and the solution is stolen from there. 😄 This is how it looks now if we have a multiline badge:
screen shot 2018-02-22 at 17 17 17

# Where should the reviewer start:
* Check the diffs and test an acceptance link.

# Unit tests:
* Added

### CODE REVIEW

**Comment by maximuk** [comment]
Better to have one property `align="left|center|right"`

File: components/badge/badge.md [hunk 1]
```diff
@@ -1,78 +1,132 @@
...
     initialState = {
-      isVisibleTop: false,
-      isVisibleRight: true,
-      isVisibleBottom: true,
-      isVisibleLeft: true,
+      align: 'start',
+      position: 'top',
       isArrow: false,
       isBorder: true,
+      isCentered: false,
```
Comment (maximuk) [inline components/badge/badge.md]: please remove the line

File: components/badge/badge.md [hunk 2]
```diff
@@ -1,78 +1,132 @@
...
       isColorBackground: false,
+      isReversed: false,
```
Comment (maximuk) [inline components/badge/badge.md]: please remove the line

File: components/badge/badge.md (hunk 1 above)
Comment (leandrooriente) [inline components/badge/badge.md]: Still valid?

File: components/badge/badge.md (hunk 2 above)
Comment (leandrooriente) [inline components/badge/badge.md]: Still valid?

File: tests/unit/badge/badge.spec.js [hunk 3]
```diff
@@ -88,4 +88,44 @@ describe('Badge', () => {
 
     expect(wrapper).toMatchSnapshot();
   });
+
+  it('should render badge with "start" alignment', () => {
```
Comment (leandrooriente) [inline tests/unit/badge/badge.spec.js]: Should we have a combined (position + align) test?

File: components/badge/badge.md (hunk 1 above)
Comment (yurist38) [reply components/badge/badge.md]: Ah, crap. Removing. Thanks!

File: tests/unit/badge/badge.spec.js (hunk 3 above)
Comment (yurist38) [reply tests/unit/badge/badge.spec.js]: @leandrooriente good idea, will add now

**Comment by AlleeX** [comment]
Hmm..

selection_266

**Comment by yurist38** [comment]
@AlleeX that's interesting. Let me check it.

**Comment by yurist38** [comment]
@AlleeX could you please check now? Should be slightly better. I found this problem explained here: https://stackoverflow.com/a/33094994

File: package.json [hunk 4]
```diff
@@ -17,6 +18,7 @@
     "prebuild": "babel --copy-files ./components --out-dir lib --ignore *.scss,*.md &",
     "styleguide-build": "styleguidist build",
     "styleguide-server": "styleguidist server",
+    "styleguide:dev": "npm run styleguide-build && npm run styleguide-server",
```
Comment (iwwwi) [inline package.json]: 👍

File: package.json [hunk 5]
```diff
@@ -5,6 +5,7 @@
   "main": "lib/index.js",
   "scripts": {
     "build:watch": "node ./builder/program -w",
+    "build:watch:theme": "node builder/program.js -t ./themes/_default.yaml -w",
```
Comment (iwwwi) [inline package.json]: 👍

File: package.json (hunk 4 above)
Comment (froskie) [reply package.json]: I'd like to namespace everything in the same way. I really like the combination of the other two, so let's try to make them all the same? Change the `-` to `:` in the other two?

File: package.json (hunk 5 above)
Comment (froskie) [reply package.json]: I didn't see the point of this one. This is what happens by default, If you look at the `builder.js` file.

**Pull request merged**